tests/core/test_test.py

```diff
@@ -390,6 +404,152 @@ def test_partial_data(sushi_context: Context) -> None:
     )
 
 
+@pytest.mark.parametrize(
+    "waiter_names_input",
+    [
+        """sushi.waiter_names:
+        format: yaml
+        rows: 
+        - id: 1
+          name: alice
+        - id: 2
+          name: 'bob'
+        """,
+        """sushi.waiter_names: 
+      format: csv
+      rows: |
+        id,name
+        1,alice
+        2,bob""",
+    ],
+)
+def test_format_inline(sushi_context: Context, waiter_names_input: str) -> None:
+    _check_successful_or_raise(
+        _create_test(
+            body=load_yaml(
+                f"""
+test_foo:
+  model: sushi.foo
+  inputs:
+    {waiter_names_input}
+  outputs:
+    query:
+      - id: 1
+        name: alice
+      - id: 2
+        name: 'bob'
+                """
+            ),
+            test_name="test_foo",
+            model=sushi_context.upsert_model(
+                _create_model(
+                    "SELECT id, name FROM sushi.waiter_names ",
+                    default_catalog=sushi_context.default_catalog,
+                )
+            ),
+            context=sushi_context,
+        ).run()
+    )
+
+
+@pytest.mark.parametrize(
+    "input_data",
+    [
+        [
+            """sushi.waiter_names:
+        format: yaml
+        path: """,
+            "test_data.yaml",
+            """- id: 1
+  name: alice
+- id: 2
+  name: 'bob'
+""",
+        ],
+        [
```

Comment:
You don't have to do this unwrapping from the list manually, `parametrize` offers a cleaner way to achieve this where you can name each part separately:

```python
    ["column", "expected"],
    [
        (  # Array of strings
            "['value1', 'value2']",
            [{"col": ["value1", "value2"]}],
        ),
```

Look into https://github.com/TobikoData/sqlmesh/blob/b11b93e1e19338b007a27c31c0f943327ca08aff/tests/core/test_test.py#L1237